File: ABC/abc412/a/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var scanner = bufio.NewScanner(os.Stdin)

func ReadLine() string {
	scanner.Scan()
	return scanner.Text()
}

func ReadInt() int {
	line := ReadLine()
	n, _ := strconv.Atoi(line)
	return n
}

func ReadInts() []int {
	line := ReadLine()
	strs := strings.Fields(line)
	ints := make([]int, len(strs))
	for i, str := range strs {
		ints[i], _ = strconv.Atoi(str)
	}
	return ints
}

func main() {
	n := ReadInt()

	count := 0
	for i := 0; i < n; i++ {
		day := ReadInts()
		target := day[0]
		done := day[1]

		if done > target {
			count++
		}
	}

	fmt.Println(count)
}

File: ABC/abc412/a/rust/src/main.rs
use std::io;

fn read_line() -> String {
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    line.trim().to_string()
}

fn read_int() -> i32 {
    read_line().parse().unwrap()
}

fn read_ints() -> Vec<i32> {
    read_line()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect()
}

fn read_2d_ints(n: usize) -> Vec<Vec<i32>> {
    (0..n).map(|_| read_ints()).collect()
}

fn main() {
    let n = read_int();
    let data = read_2d_ints(n as usize);
    let count = data.iter().filter(|values| values[1] > values[0]).count();
    println!("{}", count);
}
